<launch>
    <param name="/use_sim_time" value="false"/>
    <arg name="rviz" default="false"/>

    <!--distortion removal-->
    <include file="$(find lio_wheel)/launch/distortion_removal.launch"/>

    <!--laserMapping-->
    <rosparam command="load" file="$(find lio_wheel)/config/sjtu_config.yaml"/>

    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="2"/>
    <param name="max_iteration" type="int" value="3"/>
    <param name="filter_size_surf" type="double" value="0.5"/>
    <param name="filter_size_map" type="double" value="0.5"/>
    <param name="cube_side_length" type="double" value="1000"/>
    <param name="runtime_pos_log_enable" type="bool" value="1"/>
    <node pkg="lio_wheel" type="fastlio_mapping" name="laserMapping" output="log"/>

    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz_fast_lio"
              args="-d $(find lio_wheel)/rviz_cfg/lio_wheel.rviz"/>
    </group>

</launch>